### PR TITLE
The start of documentation 2.0

### LINKED ISSUE #3649: Outdated docs on background diffusivity/viscosity
Some of the docs still include the background viscosity in the description of Smagorinsky-Lilly. Maybe you're purposefully keeping the "Physics" different than the "Model," but if it's a typo, these are the two places it's the old version:

https://github.com/CliMA/Oceananigans.jl/blob/738d1724e5128502efe5579dd210fc17aacb9089/docs/src/physics/turbulence_closures.md?plain=1#L75-L81

https://github.com/CliMA/Oceananigans.jl/blob/738d1724e5128502efe5579dd210fc17aacb9089/docs/src/physics/turbulence_closures.md?plain=1#L104-L109

This description in the model setup is also a bit misleading, or, at least, unclear.

https://github.com/CliMA/Oceananigans.jl/blob/738d1724e5128502efe5579dd210fc17aacb9089/docs/src/model_setup/turbulent_diffusivity_closures_and_les_models.md?plain=1#L71-L72

### PR DESCRIPTION
**Pull request by glwagner**
This PR reorders the documentation and will add a `Grids` and `Fields` page to the top-level of the docs. It will also clean up minor outstanding issues.

Longer description of our intent: #3672 

Closes #3649

### CODE REVIEW

**Comment by glwagner** [comment]
@navidcy I have also enabled push previews for this PR but we can disable before merging.

**Comment by glwagner** [comment]
We'll be looking here for previews: https://clima.github.io/OceananigansDocumentation/previews/PR3673

**Comment by glwagner** [comment]
@navidcy I think it would be cool to show a visualization of the grids in the "Grids tutorial"

**Comment by francispoulin** [comment]
> We'll be looking here for previews: https://clima.github.io/OceananigansDocumentation/previews/PR3673

I tried this link but received a 404 error.  Does it work for others?

**Comment by glwagner** [comment]
> > We'll be looking here for previews: https://clima.github.io/OceananigansDocumentation/previews/PR3673
> 
> I tried this link but received a 404 error. Does it work for others?

It errors for me too! Nothing will come up until the docs build passes, looks like it is still failing.

**Comment by glwagner** [comment]
I think we should improve the docstring for `Distributed`:

https://github.com/CliMA/Oceananigans.jl/blob/8f55656812e4ad7968ff37fccdb3bd1974ef393b/src/DistributedComputations/distributed_architectures.jl#L174-L205

A few comments:

* Can we explain "synchronized communication" better? This is confusing --- I think it actually has to do with the algorithm a model uses (very far away from building the architecture). We have to explain that this concept is irrelevant unless we are using a model that has an asychronous algorithm. Or otherwise more specifically explain what this means.
* The entry for `devices` is missing a period a the end.

In general more explanation of the keyword arguments that is as local to the concept of grids as possible. The docstring makes vague references to "support for partitioning". But this refers to models, not grids. It doesn't really make sense in this context.

Also we need examples.

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:33]:
```suggestion
This simple grid:
```

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:38]:
```suggestion
* Has an ``x`` dimension that spans from `x=0`, to `x=64`, ``y`` spans from `y=0` to `y=32`, and ``z`` spans `z=0` to `z=8`.
```

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:66]: so this will fail if docs built don't run on GPUs

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:69]:
```suggestion
For the above grid, the ``y``-dimension is marked `Flat` by specifying `topology = (Periodic, Flat, Bounded)`.
This means that nothing varies in ``y``; derivatives in ``y`` are 0.
```

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:70]:
```suggestion
This also means that the keyword argument (or short, kwarg) that specifies the ``y``-domains may be omitted, and that
```

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:78]:
```suggestion
These examples illustrate how the definition of any grid requires:
```

File: docs/src/model_setup/legacy_grids.md
Comment (navidcy) [inline docs/src/model_setup/legacy_grids.md:1]: do we need this `.md` file? or are you keeping it only for now..?

File: docs/src/operations.md
Comment (navidcy) [inline docs/src/operations.md:8]:
```suggestion
and saved to disk while a simulation runs, and also for post-processing.
```

File: docs/src/simulations.md
Comment (navidcy) [inline docs/src/simulations.md:3]:
```suggestion
`Simulation`s are basically a utility for managing a time-stepping loop, including scheduling important activities such as:
```

File: src/Oceananigans.jl
Comment (navidcy) [inline src/Oceananigans.jl:22]: these were not exported? wow..

File: docs/src/grids.md
Comment (glwagner) [reply docs/src/grids.md:33]: "this simple grid: uses the CPU" isn't correct right?

File: docs/src/grids.md
Comment (glwagner) [reply docs/src/grids.md:38]: I'm referring to the kwargs here

File: docs/src/grids.md
Comment (glwagner) [reply docs/src/grids.md:66]: I changed it so the docs build on the GPU

File: docs/src/grids.md
Comment (glwagner) [reply docs/src/grids.md:78]: confused whether this is correct

File: docs/src/model_setup/legacy_grids.md
Comment (glwagner) [reply docs/src/model_setup/legacy_grids.md:1]: I'd like to keep the legacy docs around until we are 100% finished with the new docs.

File: src/Oceananigans.jl
Comment (glwagner) [reply src/Oceananigans.jl:22]: nobody is using it

File: docs/src/grids.md
Comment (glwagner) [reply docs/src/grids.md:35]: I like it

File: docs/src/grids.md
Comment (glwagner) [reply docs/src/grids.md:33]: see here:

https://style.mla.org/vertical-lists-in-mla-style/

under "Lists That Continue the Sentence Introducing Them"

File: docs/src/grids.md
Comment (navidcy) [reply docs/src/grids.md:38]: oh OK!

File: docs/src/grids.md
Comment (navidcy) [reply docs/src/grids.md:78]: ok, perhaps it's a style choice -- I'll leave it up to you to decide!

File: docs/src/model_setup/legacy_grids.md
Comment (navidcy) [reply docs/src/model_setup/legacy_grids.md:1]: makes sense -- I only flagged it in case they were forgotten there

File: docs/src/simulations.md
Comment (navidcy) [reply docs/src/simulations.md:3]: as before, perhaps it's style

File: docs/src/grids.md
Comment (glwagner) [reply docs/src/grids.md:78]: I think a colon is incorrect here!

Comment (glwagner) [reply docs/src/grids.md:78]: (because the bullet points "finish" the sentence started above)

File: docs/src/simulations.md
Comment (glwagner) [reply docs/src/simulations.md:3]: hmm no, in this case we need a colon

File: docs/src/grids.md
Comment (navidcy) [reply docs/src/grids.md:33]: great! even if it was merely your preference I'd be convinced!

File: docs/src/grids.md
Comment (navidcy) [reply docs/src/grids.md:78]: gotta

File: docs/make.jl
Comment (navidcy) [inline docs/make.jl:210]: did you check that the changes you did here do what the previous code was doing?

File: docs/src/fields.md
Comment (navidcy) [inline docs/src/fields.md:59]:
```suggestion
Another common `Field` we encounter at those with finite volumes centered over the `x`-interfaces of the primary grid,
```

File: docs/src/fields.md
Comment (navidcy) [inline docs/src/fields.md:68]:
```suggestion
The name `u` is suggestive, because within the Arakawa type C grid (or, 'C grid' for short) used by Oceananigans,
```

File: docs/src/fields.md
Comment (navidcy) [inline docs/src/fields.md:103]:
```suggestion
In a `Bounded` direction, however, the left and right end points differ and are both included among the nodes.
```

File: docs/src/fields.md
Comment (navidcy) [inline docs/src/fields.md:151]: this would do the job...

```suggestion
current_figure() # hide
```

File: docs/src/fields.md
Comment (navidcy) [inline docs/src/fields.md:323]: I felt that this section should come _before_ the bit: "For `Field`s on three-dimensional grids, the function...."

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:99]:
```suggestion
By using `architecture = GPU()`, any fields constructed on `grid` store their data on
an Nvidia [`GPU`](@ref), if one is available. By default, the grid will be constructed on
```

File: docs/src/fields.md
Comment (glwagner) [reply docs/src/fields.md]: might be good to include a reference

File: docs/src/fields.md
Comment (glwagner) [reply docs/src/fields.md:318]: Let me think about that. Also the next section is going to go into halo regions and this leads naturally into that. Maybe.

Comment (glwagner) [reply docs/src/fields.md:318]: I see what you're saying

Comment (navidcy) [reply docs/src/fields.md:318]: If we keep it as is, then perhaps half a phrase to ensure the people know that now we are connecting to the example one up... E.g. "Going back to the case where we set `c` with 2x, now, since...."

File: docs/src/fields.md
Comment (navidcy) [inline docs/src/fields.md:59]:
```suggestion
Another common type of `Field` we encounter are those with finite volumes centered over the `x`-interfaces of the primary grid,
```

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:572]: `touch` is unix command; this will work in more machines:

```suggestion
julia --project -e 'using Pkg; using Pkg; Pkg.activate("."); Pkg.add(["Oceananigans", "MPI"])'
```

Comment (glwagner) [reply docs/src/grids.md]: I think not

**Comment by glwagner** [comment]
I think we're ready to merge this once we hammer out the details. Still TODO will be:

- refactoring the model setup pages to be more tutorial-like
- finishing the tutorials on reductions, simulations, and field time series / post-processing

@navidcy @simone-silvestri @rafferrari

**Comment by navidcy** [comment]
with  `distributed_example.jl`

```
using Oceananigans
using MPI
MPI.Init()

architecture = Distributed()
on_rank_0 = MPI.Comm_rank(MPI.COMM_WORLD) == 0

if on_rank_0
    @show architecture
end

```

when I run 

```bash
(base) navid:OceananigansDistributedExamples/ (glw/docs2) $ mpiexec -n 2 julia --project distributed_example.jl
┌ Warning:     You appear to have run julia under a different `mpiexec` than the one used by MPI.jl.
│     See the documentation for details.
└ @ MPI ~/.julia/packages/MPI/is7GN/src/environment.jl:26
┌ Warning:     You appear to have run julia under a different `mpiexec` than the one used by MPI.jl.
│     See the documentation for details.
└ @ MPI ~/.julia/packages/MPI/is7GN/src/environment.jl:26
architecture = Distributed architecture (rank 0/0) [index (1, 1, 1) / (1, 1, 1)]
└── child architecture: CPU 
└── connectivity:
architecture = Distributed architecture (rank 0/0) [index (1, 1, 1) / (1, 1, 1)]
└── child architecture: CPU 
└── connectivity:

File: docs/src/simulations.md
Comment (navidcy) [inline docs/src/simulations.md:12]:
```suggestion
```

**Review by navidcy — APPROVED** [review]

**Review by navidcy — APPROVED** [review]
other than the distributed grid examples which I can't reproduce, all the rest look good to me!

**Comment by glwagner** [comment]
> with `distributed_example.jl`
> 
> ```
> using Oceananigans
> using MPI
> MPI.Init()
> 
> architecture = Distributed()
> on_rank_0 = MPI.Comm_rank(MPI.COMM_WORLD) == 0
> 
> if on_rank_0
>     @show architecture
> end
> ```
> 
> when I run
> 
> ```shell
> (base) navid:OceananigansDistributedExamples/ (glw/docs2) $ mpiexec -n 2 julia --project distributed_example.jl
> ┌ Warning:     You appear to have run julia under a different `mpiexec` than the one used by MPI.jl.
> │     See the documentation for details.
> └ @ MPI ~/.julia/packages/MPI/is7GN/src/environment.jl:26
> ┌ Warning:     You appear to have run julia under a different `mpiexec` than the one used by MPI.jl.
> │     See the documentation for details.
> └ @ MPI ~/.julia/packages/MPI/is7GN/src/environment.jl:26
> architecture = Distributed architecture (rank 0/0) [index (1, 1, 1) / (1, 1, 1)]
> └── child architecture: CPU 
> └── connectivity:
> architecture = Distributed architecture (rank 0/0) [index (1, 1, 1) / (1, 1, 1)]
> └── child architecture: CPU 
> └── connectivity:
> ```

pretty sure this means that you can't use MPI.

Can you figure out how to fix the problem, and then add that info to the docs "If you see this warning..., do this".

**Comment by simone-silvestri** [comment]
> when I run
> 
> ```shell
> (base) navid:OceananigansDistributedExamples/ (glw/docs2) $ mpiexec -n 2 julia --project distributed_example.jl
> ┌ Warning:     You appear to have run julia under a different `mpiexec` than the one used by MPI.jl.
> │     See the documentation for details.
> └ @ MPI ~/.julia/packages/MPI/is7GN/src/environment.jl:26
> ```

This warning means that you didn't configure MPI correctly to work with julia. There is some documentation on how to do it [here](https://juliaparallel.org/MPI.jl/stable/configuration/). We should add a link in the docs to this webpage

**Comment by navidcy** [comment]
> This warning means that you didn't configure MPI correctly to work with julia. There is some documentation on how to do it [here](https://juliaparallel.org/MPI.jl/stable/configuration/). We should add a link in the docs to this webpage

So I went to that doc pages but I wasn't sure -- do I need to install MPIPreferences? There is no other way?
We should definitely add a link to that docs page then.

**Comment by glwagner** [comment]
> > This warning means that you didn't configure MPI correctly to work with julia. There is some documentation on how to do it [here](https://juliaparallel.org/MPI.jl/stable/configuration/). We should add a link in the docs to this webpage
> 
> So I went to that doc pages but I wasn't sure -- do I need to install MPIPreferences? There is no other way? We should definitely add a link to that docs page then.

No the best thing to do is to call `mpiexecjl`, which is installed when you install `MPI.jl`.

**Comment by glwagner** [comment]
https://juliaparallel.org/MPI.jl/stable/reference/environment/#MPI.install_mpiexecjl

**Comment by glwagner** [comment]
This might be useful: https://juliaparallel.org/MPI.jl/stable/reference/environment/#MPICH_jll.mpiexec

File: docs/src/grids.md
Comment (navidcy) [inline docs/src/grids.md:74]: what's this?

**Review by navidcy — APPROVED** [review]

**Comment by navidcy** [comment]
The distributed doctests in `grids.md` don't run, somehow the environment fails to install?

**Comment by navidcy** [comment]
@glwagner I removed the "Distributed Grids" part from the docs. I suggest we merge this if Docs built and open a new PR with the distributed grids part of the tutorial?

**Pull request merged**
